activedata_etl/transforms/jscov_to_es.py

```diff
@@ -108,55 +164,52 @@ def process_source_file(parent_etl, count, obj, task_cluster_record):
         orphan_covered = orphan_covered - method_covered
         orphan_uncovered = orphan_uncovered - method_uncovered
 
-        new_record = set_default(
-            {
-                "test": {
-                    "name": test_name,
-                    "url": obj.testUrl
-                },
-                "source": {
-                    "language": "js",
-                    "file": file_info,
-                    "method": {
-                        "name": method_name,
-                        "covered": sorted(method_covered),
-                        "uncovered": sorted(method_uncovered),
-                        "total_covered": len(method_covered),
-                        "total_uncovered": len(method_uncovered),
-                        "percentage_covered": method_percentage_covered,
-                    }
-                },
-                "etl": {
-                    "id": count(),
-                    "source": parent_etl,
-                    "type": "join",
-                    "machine": machine_metadata,
-                    "timestamp": Date.now()
-                }
-            },
-            task_cluster_record
-        )
-        key = etl2key(new_record.etl)
-        yield {"id": key, "value": new_record}
+        # Record method coverage info
+        record["source"]["method"] = {
+            "name": method_name,
+            "covered": sorted(method_covered),
+            "uncovered": sorted(method_uncovered),
+            "total_covered": len(method_covered),
+            "total_uncovered": len(method_uncovered),
+            "percentage_covered": method_percentage_covered,
+        }
+
+        # Timestamp this record
+        record["etl"]["timestamp"] = Date.now()
+
+        key = etl2key(record.etl)
+        yield {"id": key, "value": record}
 
     # a record for all the lines that are not in any method
     # every file gets one because we can use it as canonical representative
+    # Record method coverage info
+    record["source"]["method"] = {
```

Comment:
`record` is a `Data` object, not a `dict`.  Instead of 

    record["source"]["method"] = {

you can write

    record.source.method = {

with the added benefit that the necessary objects will be created if they are missing